code/modules/randomMaps/vaults.dm

```diff
@@ -95,6 +95,13 @@
 
 	message_admins("<span class='info'>Loaded [result] out of [surprise_number] mining surprises.</span>")
 
+/proc/generate_hoboshack()
+	var/list/list_of_shacks = get_map_element_objects(/datum/map_element/hoboshack)
+
+	var/result = populate_area_with_vaults(/area/mine/unexplored, list_of_shacks, 1, filter_function=/proc/asteroid_can_be_placed)
```

Comment:
I may be mistaken, but won't this filter function place most shacks in the middle of rock, making it hard for hobos to escape?

Reply:
They have a mining pickaxe and other tools in their shacks to help them get out